pkg/pipeline/datasource_project_test.go

```diff
@@ -0,0 +1,85 @@
+package pipeline_test
+
+import (
+	"fmt"
+	"math/rand"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/jfrog/terraform-provider-pipeline/pkg/acctest"
+	"github.com/jfrog/terraform-provider-shared/test"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+func TestAccDatasourceProject(t *testing.T) {
+	var integrationId int
+
+	rand.Seed(time.Now().UnixNano())
+	projectKey := fmt.Sprintf("t%d", test.RandomInt())
+	integrationName := fmt.Sprintf("int%d", test.RandomInt())
+	_, fqrn, name := test.MkNames(projectKey, "pipeline_project")
+	dataSourceName := fmt.Sprintf("data.%s", fqrn)
+
+	config := util.ExecuteTemplate("TestDatasourceProjectConfig", `
+		data "pipeline_project" "{{ .name }}" {
+			name = "{{ .projectKey }}"
+		}
+	`, map[string]interface{}{
+		"name":       name,
+		"projectKey": projectKey,
+	})
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(t)
+			acctest.CreateProject(t, projectKey)
+			integrationId = acctest.CreateProjectIntegration(t, integrationName, projectKey)
```

Comment:
The project is not considered associated with a Pipeline until it's attached to an integration, node pool, or source. Hence I need to create this integration so that the data source will return project data.